### ISSUE TITLE
Docker compose shows that init files are ran, but table aren't present in the database

### ISSUE THREAD

**Issue by MathieuSauce** [gvenzl/oci-oracle-xe#169]
Hi

I'm using your oracle image in a docker compose setup, but i'm running into an issue that i wasn't able to find solution to after searching the whole internet.

Upon using docker compose to run your image, the logs show that my init script that was placed in /container-entrypoint-initdb.d is ran, but i can't seem to find my tables in the database that is exposed after the container is up and running, as if that script never existed.

I tried running a test-script that would SELECT from said table right after init, and of course it would work. But after the database is done initializing, it seems my tables are deleted from the db.

Here's my docker-compose file

```
services:
  oracle:
    image: docker.artifactory-dogen.group.echonet/gvenzl/oracle-xe
    ports: 
      - "1521:1521"      
    environment:
      ORACLE_PASSWORD : test
    volumes:
      - ./init.sql:/container-entrypoint-initdb.d/init.sql
      
```

I'm using sqlplus to connect to the localhost/xe database in the container, using system/test as login.

I'm pretty sure i don't fully understand the tool, since i'm pretty new to it, and my mistake might be pretty dumb.

Thank you in advance

**Comment by mprins** [comment]
are you looking at the right database? please show us your connect string

**Comment by MathieuSauce** [comment]
I have tried both 
``` sqlplus system/test@//localhost/XEPDB1```
And
``` sqlplus system/test@//localhost/XE```

I have also tried creating another db using the ORACLE_DATABASE variable and an APP_USER, but those didn't work either

**Comment by akridge** [comment]
Not sure if we had a similar issue, but I noticed this in the readme that helped solve things for me:
- "The *.sql, *.sql.gz and *.sql.zip files will be executed in SQL*Plus as the SYS user connected to the Oracle instance (XE). "

I was able to solve my issue by adding the following at the start of my install sql script:
```
ALTER SESSION SET CONTAINER=XEPDB1;
```
and the following at the end
```
commit;
```

**Comment by gvenzl** [comment]
Hey @MathieuSauce,

Can you share your `init.sql` and the log output of the container, please?

**Comment by proteintogo** [comment]
I'm facing the same issue while using TestContainer in a SpringBoot Application

```
static OracleContainer oracle = new OracleContainer("gvenzl/oracle-xe:21-slim-faststart")
			.withUsername("FO_IAM").withPassword("FO_IAM")
			.withCopyFileToContainer(MountableFile.forHostPath("init_script.sql"), "/container-entrypoint-initdb.d/init.sql");
```

The init_script.sql is placed under resources folder and it looks like this:
```
ALTER SESSION set container=xepdb1;

-- NOMENCLATURES
CREATE USER nomenclatures
    IDENTIFIED BY nomenclatures
    DEFAULT TABLESPACE USERS
    TEMPORARY TABLESPACE TEMP
    PROFILE DEFAULT
    ACCOUNT UNLOCK;
GRANT CONNECT TO nomenclatures;
GRANT RESOURCE TO nomenclatures;
ALTER USER nomenclatures DEFAULT ROLE ALL;
GRANT CREATE ANY DIRECTORY TO nomenclatures;
GRANT CREATE JOB TO nomenclatures;
GRANT CREATE MATERIALIZED VIEW TO nomenclatures;
GRANT CREATE PROCEDURE TO nomenclatures;
GRANT CREATE SEQUENCE TO nomenclatures;
GRANT CREATE SYNONYM TO nomenclatures;
GRANT CREATE TABLE TO nomenclatures;
GRANT CREATE TRIGGER TO nomenclatures;
GRANT CREATE TYPE TO nomenclatures;
GRANT CREATE VIEW TO nomenclatures;
GRANT DEBUG CONNECT SESSION TO nomenclatures;
GRANT QUERY REWRITE TO nomenclatures;
GRANT UNLIMITED TABLESPACE TO nomenclatures;
ALTER USER nomenclatures QUOTA 100M on USERS;

-- FO_IAM
CREATE USER fo_iam
    IDENTIFIED BY fo_iam
    DEFAULT TABLESPACE USERS
    TEMPORARY TABLESPACE TEMP
    PROFILE DEFAULT
    ACCOUNT UNLOCK;
GRANT CONNECT TO fo_iam;
GRANT RESOURCE TO fo_iam;
ALTER USER fo_iam DEFAULT ROLE ALL;
GRANT CREATE ANY DIRECTORY TO fo_iam;
GRANT CREATE JOB TO fo_iam;
GRANT CREATE MATERIALIZED VIEW TO fo_iam;
GRANT CREATE PROCEDURE TO fo_iam;
GRANT CREATE SEQUENCE TO fo_iam;
GRANT CREATE SYNONYM TO fo_iam;
GRANT CREATE TABLE TO fo_iam;
GRANT CREATE TRIGGER TO fo_iam;
GRANT CREATE TYPE TO fo_iam;
GRANT CREATE VIEW TO fo_iam;
GRANT DEBUG CONNECT SESSION TO fo_iam;
GRANT QUERY REWRITE TO fo_iam;
GRANT UNLIMITED TABLESPACE TO fo_iam;
ALTER USER fo_iam quota 100M on USERS;


-- Connect as the new user
ALTER SESSION SET CURRENT_SCHEMA=nomenclatures;

-- Create the country table
CREATE TABLE NOMENCLATURES.COUNTRIES
(
    pk_country   NUMBER PRIMARY KEY,
    english_short_name varchar2(128 char),
    alpha3_code VARCHAR2(50 char)
);

GRANT SELECT ON NOMENCLATURES.COUNTRIES TO FO_IAM;

COMMIT;
```

The FO_IAM schema is created but I think that this is happening thanks to `.withUsername()` method.
The logs looks good to me. Any hint why nomenclatures schema is not created?
```
2023-02-06 21:47:21 Database opened.
2023-02-06 21:47:21 
2023-02-06 21:47:21 CONTAINER: Resetting SYS and SYSTEM passwords.
2023-02-06 21:47:21 
2023-02-06 21:47:21 User altered.
2023-02-06 21:47:21 
2023-02-06 21:47:21 
2023-02-06 21:47:21 User altered.
2023-02-06 21:47:21 
2023-02-06 21:47:21 
2023-02-06 21:47:21 Session altered.
2023-02-06 21:47:21 
2023-02-06 21:47:21 
2023-02-06 21:47:21 User created.
2023-02-06 21:47:21 
2023-02-06 21:47:21 
2023-02-06 21:47:21 Grant succeeded.
2023-02-06 21:47:21 
2023-02-06 21:47:21 
2023-02-06 21:47:21 CONTAINER: Executing user defined scripts...
2023-02-06 21:47:21 
2023-02-06 21:47:21 CONTAINER: running /container-entrypoint-initdb.d/init.sql ...
2023-02-06 21:47:21 CONTAINER: DONE: running /container-entrypoint-initdb.d/init.sql
2023-02-06 21:47:21 
2023-02-06 21:47:21 CONTAINER: DONE: Executing user defined scripts.
2023-02-06 21:47:21 
2023-02-06 21:47:21 
2023-02-06 21:47:21 #########################
2023-02-06 21:47:21 DATABASE IS READY TO USE!
2023-02-06 21:47:21 #########################
```

**Comment by adamalexandru4** [comment]
I'm facing the same issue while using TestContainer in a SpringBoot Application

```
static OracleContainer oracle = new OracleContainer("gvenzl/oracle-xe:21-slim-faststart")
			.withUsername("FO_IAM").withPassword("FO_IAM")
			.withCopyFileToContainer(MountableFile.forHostPath("init_script.sql"), "/container-entrypoint-initdb.d/init.sql");
```

The init_script.sql is placed under resources folder and it looks like this:
```
ALTER SESSION set container=xepdb1;

-- NOMENCLATURES
CREATE USER nomenclatures
    IDENTIFIED BY nomenclatures
    DEFAULT TABLESPACE USERS
    TEMPORARY TABLESPACE TEMP
    PROFILE DEFAULT
    ACCOUNT UNLOCK;
GRANT CONNECT TO nomenclatures;
GRANT RESOURCE TO nomenclatures;
ALTER USER nomenclatures DEFAULT ROLE ALL;
GRANT CREATE ANY DIRECTORY TO nomenclatures;
GRANT CREATE JOB TO nomenclatures;
GRANT CREATE MATERIALIZED VIEW TO nomenclatures;
GRANT CREATE PROCEDURE TO nomenclatures;
GRANT CREATE SEQUENCE TO nomenclatures;
GRANT CREATE SYNONYM TO nomenclatures;
GRANT CREATE TABLE TO nomenclatures;
GRANT CREATE TRIGGER TO nomenclatures;
GRANT CREATE TYPE TO nomenclatures;
GRANT CREATE VIEW TO nomenclatures;
GRANT DEBUG CONNECT SESSION TO nomenclatures;
GRANT QUERY REWRITE TO nomenclatures;
GRANT UNLIMITED TABLESPACE TO nomenclatures;
ALTER USER nomenclatures QUOTA 100M on USERS;

-- FO_IAM
CREATE USER fo_iam
    IDENTIFIED BY fo_iam
    DEFAULT TABLESPACE USERS
    TEMPORARY TABLESPACE TEMP
    PROFILE DEFAULT
    ACCOUNT UNLOCK;
GRANT CONNECT TO fo_iam;
GRANT RESOURCE TO fo_iam;
ALTER USER fo_iam DEFAULT ROLE ALL;
GRANT CREATE ANY DIRECTORY TO fo_iam;
GRANT CREATE JOB TO fo_iam;
GRANT CREATE MATERIALIZED VIEW TO fo_iam;
GRANT CREATE PROCEDURE TO fo_iam;
GRANT CREATE SEQUENCE TO fo_iam;
GRANT CREATE SYNONYM TO fo_iam;
GRANT CREATE TABLE TO fo_iam;
GRANT CREATE TRIGGER TO fo_iam;
GRANT CREATE TYPE TO fo_iam;
GRANT CREATE VIEW TO fo_iam;
GRANT DEBUG CONNECT SESSION TO fo_iam;
GRANT QUERY REWRITE TO fo_iam;
GRANT UNLIMITED TABLESPACE TO fo_iam;
ALTER USER fo_iam quota 100M on USERS;


-- Connect as the new user
ALTER SESSION SET CURRENT_SCHEMA=nomenclatures;

-- Create the country table
CREATE TABLE NOMENCLATURES.COUNTRIES
(
    pk_country   NUMBER PRIMARY KEY,
    english_short_name varchar2(128 char),
    alpha3_code VARCHAR2(50 char)
);

GRANT SELECT ON NOMENCLATURES.COUNTRIES TO FO_IAM;

COMMIT;
```

The FO_IAM schema is created but I think that this is happening thanks to `.withUsername()` method.
The logs looks good to me. Any hint why nomenclatures schema is not created?
```
2023-02-06 21:47:21 Database opened.
2023-02-06 21:47:21 
2023-02-06 21:47:21 CONTAINER: Resetting SYS and SYSTEM passwords.
2023-02-06 21:47:21 
2023-02-06 21:47:21 User altered.
2023-02-06 21:47:21 
2023-02-06 21:47:21 
2023-02-06 21:47:21 User altered.
2023-02-06 21:47:21 
2023-02-06 21:47:21 
2023-02-06 21:47:21 Session altered.
2023-02-06 21:47:21 
2023-02-06 21:47:21 
2023-02-06 21:47:21 User created.
2023-02-06 21:47:21 
2023-02-06 21:47:21 
2023-02-06 21:47:21 Grant succeeded.
2023-02-06 21:47:21 
2023-02-06 21:47:21 
2023-02-06 21:47:21 CONTAINER: Executing user defined scripts...
2023-02-06 21:47:21 
2023-02-06 21:47:21 CONTAINER: running /container-entrypoint-initdb.d/init.sql ...
2023-02-06 21:47:21 CONTAINER: DONE: running /container-entrypoint-initdb.d/init.sql
2023-02-06 21:47:21 
2023-02-06 21:47:21 CONTAINER: DONE: Executing user defined scripts.
2023-02-06 21:47:21 
2023-02-06 21:47:21 
2023-02-06 21:47:21 #########################
2023-02-06 21:47:21 DATABASE IS READY TO USE!
2023-02-06 21:47:21 #########################
```

**Comment by akridge** [comment]
@adamalexandru4 If its helpful for your debugging, I copied your "init_script.sql" code and had no issues running it via docker compose. The users, table and everything were created.

**Comment by gvenzl** [comment]
Hey @adamalexandru4,

Your file appears to have not contained anything, you should see the output of every command that's in your script.
This is what I get when running your init script, note all the output from the script after `CONTAINER: running /container-entrypoint-initdb.d/init_script.sql ...`:

```
$ cat init_script.sql
ALTER SESSION set container=xepdb1;

-- NOMENCLATURES
CREATE USER nomenclatures
    IDENTIFIED BY nomenclatures
    DEFAULT TABLESPACE USERS
    TEMPORARY TABLESPACE TEMP
    PROFILE DEFAULT
    ACCOUNT UNLOCK;
GRANT CONNECT TO nomenclatures;
GRANT RESOURCE TO nomenclatures;
ALTER USER nomenclatures DEFAULT ROLE ALL;
GRANT CREATE ANY DIRECTORY TO nomenclatures;
GRANT CREATE JOB TO nomenclatures;
GRANT CREATE MATERIALIZED VIEW TO nomenclatures;
GRANT CREATE PROCEDURE TO nomenclatures;
GRANT CREATE SEQUENCE TO nomenclatures;
GRANT CREATE SYNONYM TO nomenclatures;
GRANT CREATE TABLE TO nomenclatures;
GRANT CREATE TRIGGER TO nomenclatures;
GRANT CREATE TYPE TO nomenclatures;
GRANT CREATE VIEW TO nomenclatures;
GRANT DEBUG CONNECT SESSION TO nomenclatures;
GRANT QUERY REWRITE TO nomenclatures;
GRANT UNLIMITED TABLESPACE TO nomenclatures;
ALTER USER nomenclatures QUOTA 100M on USERS;

-- FO_IAM
CREATE USER fo_iam
    IDENTIFIED BY fo_iam
    DEFAULT TABLESPACE USERS
    TEMPORARY TABLESPACE TEMP
    PROFILE DEFAULT
    ACCOUNT UNLOCK;
GRANT CONNECT TO fo_iam;
GRANT RESOURCE TO fo_iam;
ALTER USER fo_iam DEFAULT ROLE ALL;
GRANT CREATE ANY DIRECTORY TO fo_iam;
GRANT CREATE JOB TO fo_iam;
GRANT CREATE MATERIALIZED VIEW TO fo_iam;
GRANT CREATE PROCEDURE TO fo_iam;
GRANT CREATE SEQUENCE TO fo_iam;
GRANT CREATE SYNONYM TO fo_iam;
GRANT CREATE TABLE TO fo_iam;
GRANT CREATE TRIGGER TO fo_iam;
GRANT CREATE TYPE TO fo_iam;
GRANT CREATE VIEW TO fo_iam;
GRANT DEBUG CONNECT SESSION TO fo_iam;
GRANT QUERY REWRITE TO fo_iam;
GRANT UNLIMITED TABLESPACE TO fo_iam;
ALTER USER fo_iam quota 100M on USERS;


-- Connect as the new user
ALTER SESSION SET CURRENT_SCHEMA=nomenclatures;

-- Create the country table
CREATE TABLE NOMENCLATURES.COUNTRIES
(
    pk_country   NUMBER PRIMARY KEY,
    english_short_name varchar2(128 char),
    alpha3_code VARCHAR2(50 char)
);

GRANT SELECT ON NOMENCLATURES.COUNTRIES TO FO_IAM;

COMMIT;

$ podman run --name test -e ORACLE_RANDOM_PASSWORD=y -v /home/gvenzl/test:/container-entrypoint-initdb.d/ gvenzl/oracle-xe:21-slim-faststart
CONTAINER: starting up...
CONTAINER: first database startup, initializing...
CONTAINER: starting up Oracle Database...

LSNRCTL for Linux: Version 21.0.0.0.0 - Production on 07-FEB-2023 02:25:21

Copyright (c) 1991, 2021, Oracle.  All rights reserved.

Starting /opt/oracle/product/21c/dbhomeXE/bin/tnslsnr: please wait...

TNSLSNR for Linux: Version 21.0.0.0.0 - Production
System parameter file is /opt/oracle/homes/OraDBHome21cXE/network/admin/listener.ora
Log messages written to /opt/oracle/diag/tnslsnr/4d81bd054ab0/listener/alert/log.xml
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC_FOR_XE)))
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))

Connecting to (DESCRIPTION=(ADDRESS=(PROTOCOL=IPC)(KEY=EXTPROC_FOR_XE)))
STATUS of the LISTENER
------------------------
Alias                     LISTENER
Version                   TNSLSNR for Linux: Version 21.0.0.0.0 - Production
Start Date                07-FEB-2023 02:25:21
Uptime                    0 days 0 hr. 0 min. 0 sec
Trace Level               off
Security                  ON: Local OS Authentication
SNMP                      OFF
Default Service           XE
Listener Parameter File   /opt/oracle/homes/OraDBHome21cXE/network/admin/listener.ora
Listener Log File         /opt/oracle/diag/tnslsnr/4d81bd054ab0/listener/alert/log.xml
Listening Endpoints Summary...
  (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC_FOR_XE)))
  (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))
The listener supports no services
The command completed successfully
ORACLE instance started.

Total System Global Area 1241512272 bytes
Fixed Size		    9685328 bytes
Variable Size		  637534208 bytes
Database Buffers	  587202560 bytes
Redo Buffers		    7090176 bytes
Database mounted.
Database opened.

CONTAINER: Resetting SYS and SYSTEM passwords.

User altered.


User altered.

############################################
ORACLE PASSWORD FOR SYS AND SYSTEM: ODRlZjQ0
############################################

CONTAINER: Executing user defined scripts...

CONTAINER: running /container-entrypoint-initdb.d/init_script.sql ...

Session altered.


User created.


Grant succeeded.


Grant succeeded.


User altered.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


User altered.


User created.


Grant succeeded.


Grant succeeded.


User altered.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


Grant succeeded.


User altered.


Session altered.


Table created.


Grant succeeded.


Commit complete.

CONTAINER: DONE: running /container-entrypoint-initdb.d/init_script.sql

CONTAINER: DONE: Executing user defined scripts.


#########################
DATABASE IS READY TO USE!
#########################
```

**Comment by gvenzl** [comment]
Closing due to lack of response. Please reopen if you are still facing this issue.
